src/api/FileActivities.js

```diff
@@ -0,0 +1,106 @@
+    const hasActivityTypes = !!activityTypes && !!activityTypes.length;
+    const enabledRepliesQueryParam = enableReplies ? `&enable_replies=true&reply_limit=${REPLY_LIMIT}` : '';
+    const activityTypeQueryParam = hasActivityTypes ? `&activity_types=${activityTypes.join()}` : '';
+
+    return `${baseEndpoint}${activityTypeQueryParam}${enabledRepliesQueryParam}`;
+};
+
+class FileActivities extends Base {
+    /**
+     * API URL for file activities
+     *
+     * @param {string} [id] - a box file id
+     * @return {string} base url for files
+     */
+    getUrl(id?: string): string {
+        if (!id) {
+            throw new Error('Missing file id!');
+        }
+
+        return `${this.getBaseApiUrl()}${getFileActivityQueryParams(id, undefined, true)}`;
+    }
+
+    /**
+     * API URL for filtered file activities
+     *
+     * @param {string} [id] - a box file id
+     * @param {Array<FileActivityTypes>} activityTypes - optional. Array of File Activity types to filter by, returns all Activity Types if omitted.
+     * @return {string} base url for files
+     */
+    getFilteredUrl(id?: string, activityTypes?: FileActivityTypes[]): string {
+        if (!id) {
+            throw new Error('Missing file id!');
+        }
+
+        return `${this.getBaseApiUrl()}${getFileActivityQueryParams(id, activityTypes, true)}`;
+    }
+
+    /**
+     * API for fetching file activities
+     *
+     * @param {Array<FileActivityTypes>} activityTypes - optional. Array of File Activity types to filter by, returns all Activity Types if omitted.
+     * @param {Function} errorCallback - the error callback
+     * @param {string} fileId - the file id
+     * @param {BoxItemPermission} permissions - the permissions for the file
+     * @param {number} repliesCount - number of replies to return, by deafult all replies all returned
+     * @param {Function} successCallback - the success callback
+     * @returns {void}
+     */
+    getActivities({
+        activityTypes,
+        errorCallback,
+        fileId,
+        permissions,
+        repliesCount,
+        successCallback,
+    }: {
+        activityTypes: FileActivityTypes[],
+        errorCallback: (e: ElementsXhrError, code: string) => void,
+        fileId: string,
+        permissions: BoxItemPermission,
+        repliesCount?: number,
+        successCallback: (activity: FileActivity) => void,
+    }): void {
+        this.errorCode = ERROR_CODE_FETCH_ACTIVITY;
+        try {
+            this.checkApiCallValidity(PERMISSION_CAN_COMMENT, permissions, fileId);
+            this.checkApiCallValidity(PERMISSION_CAN_VIEW_ANNOTATIONS, permissions, fileId);
+        } catch (e) {
+            errorCallback(e, this.errorCode);
+            return;
+        }
+
+        this.get({
+            id: fileId,
+            successCallback,
+            errorCallback,
+            requestData: {
+                ...(repliesCount ? { replies_count: repliesCount } : null),
```

Comment:
I believe i've used `&&` before but a little hesitant to use as its not as explicit and i've had unexpected results in the past. In this case it would probably work, but I think if theres no improvement, Id rather use ternary.